frontend/src/containers/PublicGroup.js

```diff
@@ -201,28 +200,28 @@ export class PublicGroup extends Component {
   }
 
   componentDidMount() {
+    const promises = [];
     const {
+      slug,
+      transactions,
       group,
       fetchTransactions,
       fetchUsers,
       fetchGroup
     } = this.props;
 
-    return Promise.all([
-      fetchGroup(group.id),
-      fetchTransactions(group.id, FETCH_DONATIONS_OPTIONS),
-      fetchTransactions(group.id, FETCH_EXPENSES_OPTIONS),
-      fetchUsers(group.id)
-    ])
+    if (!group.name) promises.push(fetchGroup(slug));
+    if (!group.usersByRole) promises.push(fetchUsers(slug));
+    if (!transactions) promises.push(fetchTransactions(slug, FETCH_DONATIONS_OPTIONS));
+    if (!transactions) promises.push(fetchTransactions(slug, FETCH_EXPENSES_OPTIONS));
```

Comment:
gotcha. Well, I got rid of `fetchTransactions` anyway in the coming PR, so let's ignore this for now.